apps/builder/app/shared/instance-utils.test.tsx

```diff
@@ -3165,6 +3171,101 @@ describe("insert webstudio fragment copy", () => {
       availableDataSources: new Set(),
     });
     expect(data.instances.size).toEqual(4);
-    expect(newInstanceIds.size).toEqual(4);
+    expect(newInstanceIds.size).toEqual(5);
+  });
+});
+
+describe("support copy paste of :root styles", () => {
+  const css = (strings: TemplateStringsArray, ...values: string[]) => {
+    let css = "";
+    strings.forEach((string, i) => {
+      css += string + (values[i] || "");
+    });
+    const styleSourceSelections: StyleSourceSelections = new Map();
+    const styleSources: StyleSources = new Map();
+    const styles: Styles = new Map();
+    const parsed = mapGroupBy(
+      parseCss(css),
+      (parsedStyleDecl) => parsedStyleDecl.selector
+    );
+    for (const [selector, parsedStyles] of parsed) {
+      const [instanceId, styleSourceId] = selector.split("__");
+      styleSourceSelections.set(instanceId, {
+        instanceId: instanceId,
+        values: [styleSourceId],
+      });
+      styleSources.set(styleSourceId, { id: styleSourceId, type: "local" });
+      for (const { property, value } of parsedStyles) {
+        const styleDecl: StyleDecl = {
+          breakpointId: "baseId",
+          styleSourceId,
+          property,
+          value,
+        };
+        styles.set(getStyleDeclKey(styleDecl), styleDecl);
+      }
+    }
+    return { styleSourceSelections, styleSources, styles };
+  };
+
+  test("should add :root local styles", () => {
+    const oldProject = getWebstudioDataStub({
+      ...renderJsx(<$.Body ws:id="oldProjectBodyId"></$.Body>),
+      ...css`
+        :root__oldprojectlocalid {
```

Comment:
hmm, why do we need this second part starting with underscores, I wonder if there is a better way, don't want this to be user facing at some point

Reply:
doesn't matter, this is figure